### DISCUSSION TITLE
How can we access and change field values programmatically?

### DISCUSSION THREAD

**Discussion by Optiq01** [payloadcms/payload#322] [Q&A] [answered]
I have a field group that I want to toggle on and off with a checkbox.  Coming from an Angular background I'm making all my fields as their own individual files to import and orchestrate together where needed so none of their instances of use are hardcoded.  How can I listen to the value of the checkbox and dig down into the imported field group and change the 'hidden' property's value accordingly?  Is this something we can currently do?

**Comment by jmikrut** [comment, marked as answer]
Hey @Optiq01 — great question!

There are a _**lot**_ of ways that this could be accomplished. Firstly, have you seen the [Conditional Logic](https://payloadcms.com/docs/fields/overview#conditional-logic) feature yet? Your use case of having a checkbox to toggle a field set on and off is exactly why we built it. 

See this screen capture for a similar implementation:

https://user-images.githubusercontent.com/6651896/134675469-fc4d0794-ebfe-4c21-9641-42e36d9cf38c.mp4

Now, this feature only shows and hides field groups. Alternatively, if you need to _**change the values**_ of existing fields when some other field is modified, then you can use either a field hook, which allows you to set a field's value dynamically based on other field values, or you can create a custom field component and utilize our forms' hooks, which gives you access to both send and receive values of any field in your form, wherever you want.

Here is another discussion that references this:

https://github.com/payloadcms/payload/discussions/236#discussioncomment-1038883

**Reply by Optiq01** [reply]
Ahhh splendid, and yes I did see that part of the docs however I was confused about how to access a sibling's data and going back and forth between that and the hooks in terms of figuring out how it could be done.  I've also been studying the repository and come to dead ends where everything points at your dist folder which isn't publicly available.  The docs are excellent but I'm not smart enough for things to be as self explanatory as they might be for others lol.  I tinkered around and got it to work but you mentioned there's lots of ways this can be done.  What are the others?
